buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy

```diff
@@ -29,26 +31,41 @@ class VagrantTestPlugin implements Plugin<Project> {
             'sles-12',
             'ubuntu-1404',
             'ubuntu-1604'
-    ]
+    ])
+
+    /** All Windows boxes that we test, which may or may not be supplied **/
+    static final List<String> WINDOWS_BOXES = unmodifiableList([
+            'windows-2012r2',
+            'windows-2016'
+    ])
+
+    /** All boxes that we test, some of which may not be supplied **/
+    static final List<String> ALL_BOXES = unmodifiableList(LINUX_BOXES + WINDOWS_BOXES)
 
     /** Boxes used when sampling the tests **/
-    static List<String> SAMPLE = [
+    static final List<String> SAMPLE = unmodifiableList([
             'centos-7',
-            'ubuntu-1404',
-    ]
+            'ubuntu-1404'
+    ])
+
+    /** Boxes that have been supplied and are available for testing **/
+    List<String> AVAILABLE_BOXES = []
```

Comment:
This and the env vars should be camelCase since they are not final.

Reply:
Can you put the non-final stuff below the final stuff? Mixing them up bothers me more than it should.

Reply:
Agree, makes it much more clear how it works